comptoir-ejb/src/main/java/be/valuya/comptoir/service/SaleService.java

```diff
@@ -371,19 +371,19 @@ public ItemSale saveItemSale(ItemSale itemSale) {
         Price price = itemSale.getPrice();
         Item item = itemSale.getItem();
         Price itemPrice = item.getCurrentPrice();
+        
         if (price == null) {
+            BigDecimal vatExclusive = itemPrice.getVatExclusive();
+            BigDecimal vatRate = itemPrice.getVatRate();
+            // TODO: create a new price if necessary... or clean up on sale save?
             price = new Price();
+            price.setVatExclusive(vatExclusive);
+            price.setVatRate(vatRate);
         }
 
         SalePrice salePrice = AccountingUtils.calcSalePrice(price, quantity);
 
-        BigDecimal vatExclusive = itemPrice.getVatExclusive();
         BigDecimal total = salePrice.getBase();
-        BigDecimal vatRate = itemPrice.getVatRate();
-
-        // TODO: create a new price if necessary... or clean up on sale save?
-        price.setVatExclusive(vatExclusive);
-        price.setVatRate(vatRate);
 
```

Comment:
Fix pushed, see other tickets.

Le mar. 25 août 2015 à 11:07, cghislai <notifications@github.com> a écrit :

> In comptoir-ejb/src/main/java/be/valuya/comptoir/service/SaleService.java
> <https://github.com/Valuya/comptoir/pull/55#discussion_r37846383>:
>
> >          BigDecimal total = salePrice.getBase();
> > -        BigDecimal vatRate = itemPrice.getVatRate();
> > -
> > -        // TODO: create a new price if necessary... or clean up on sale save?
> > -        price.setVatExclusive(vatExclusive);
> > -        price.setVatRate(vatRate);
> >
>
> I still left it this way to prevent overwriting. Should the price==null
> check be removed?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Valuya/comptoir/pull/55/files#r37846383>.
>
-- 
Yannick Majoros
Valuya sprl